File: appkit/split_view.m
#import <AppKit/AppKit.h>
#import "split_view.h"

bool SplitView_ArrangesAllSubviews(void* ptr) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView arrangesAllSubviews];
}

void SplitView_SetArrangesAllSubviews(void* ptr, bool arrangesAllSubviews) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	[splitView setArrangesAllSubviews:arrangesAllSubviews];
}

bool SplitView_IsVertical(void* ptr) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView isVertical];
}

void SplitView_SetVertical(void* ptr, bool vertical) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	[splitView setVertical:vertical];
}

void* SplitView_DividerColor(void* ptr) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView dividerColor];
}

double SplitView_DividerThickness(void* ptr) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView dividerThickness];
}

long SplitView_DividerStyle(void* ptr) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView dividerStyle];
}

void SplitView_SetDividerStyle(void* ptr, long dividerStyle) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	[splitView setDividerStyle:dividerStyle];
}

void* SplitView_NewSplitView(NSRect frame) {
	NSSplitView* splitView = [NSSplitView alloc];
	return [[splitView initWithFrame:frame] autorelease];
}

double SplitView_MinPossiblePositionOfDividerAtIndex(void* ptr, long dividerIndex) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView minPossiblePositionOfDividerAtIndex:dividerIndex];
}

double SplitView_MaxPossiblePositionOfDividerAtIndex(void* ptr, long dividerIndex) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	return [splitView maxPossiblePositionOfDividerAtIndex:dividerIndex];
}

void SplitView_SetPosition(void* ptr, double position, long dividerIndex) {
	NSSplitView* splitView = (NSSplitView*)ptr;
	[splitView setPosition:position ofDividerAtIndex:dividerIndex];
}

File: scripts/split_view.py
#!env python3

from generate import Component, Property, InitMethod, Param, Method, Return

if __name__ == "__main__":
    w = Component(
        Type="appkit.SplitView",
        super_type='appkit.View',
        description="a view that arranges two or more views in a linear stack running horizontally or vertically",
        init_method=InitMethod(name="initWithFrame", params=[Param(name='frame', Type='foundation.Rect')]),
        properties=[
            Property(name='arrangesAllSubviews', Type='bool', getter_prefix_is=False,
                     description='whether the split view arranges all of its subviews as split panes'),
            Property(name='vertical', Type='bool',
                     description='the geometric orientation of the split view\'s dividers'),
            Property(name='dividerColor', Type='Color', readonly=True,
                     description='the color of the dividers that the split view draws between subviews'),
            Property(name='dividerThickness', Type='float64', readonly=True,
                     description='the thickness of the dividers for the split view'),
            Property(name='dividerStyle', Type='int', go_alias_type='SplitViewDividerStyle',
                     description='the style of divider between views'),
        ],
        methods=[
            Method(
                name='minPossiblePositionOfDividerAtIndex',
                params=[Param(name='dividerIndex', Type='int')],
                return_value=Return('float64'),
                description='returns the minimum possible position of the divider at the specified index'
            ),
            Method(
                name='maxPossiblePositionOfDividerAtIndex',
                params=[Param(name='dividerIndex', Type='int')],
                return_value=Return('float64'),
                description='returns the maximum possible position of the divider at the specified index'
            ),
            Method(
                name='setPosition',
                params=[
                    Param(name='position', Type='float64'),
                    Param(name='dividerIndex', Type='int', objc_param_name='ofDividerAtIndex'),
                ],
                description='sets the position of the divider at the specified index'
            ),
        ]
    )
    w.generate_code()
